### PR TITLE
Renaming sale into invoice

### PR DESCRIPTION
**Pull request by DedrickEnc**
This PR is renaming sale into invoice, here are things that I did to perform this task : 
- Cleaning the database schema by removing unused table or columns (the table beneficiary, client consumption_rummage, and sale column discount are removed)
- Updating sale name into invoice name every where (table, columns, js file, html file, comments if necessary)
- The big file ledger is unused in bhima-2.X so it is not updated, it can be removed when necessary (many routes in our server are tailed to this file)
- `sales_record.js` and `sales_record.html` are removed.

@jniles @sfount 

Thank you for contributing!

Before submitting this pull request, please verify that you have:
- [ ] Run your code through JSHint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
- [x] Run integration tests.
- [x] Run end-to-end tests.
- [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -34,3 +34,7 @@ integrations.log
 #ignore the tmp file
 
 /utilities/tmp/
+
+#ignore webstorm files
```
Comment (sfount) [inline .gitignore:38]: :+1:

File: server/controllers/finance/sale.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // DEPRECIATED - TO BE REMOVED
```
Comment (jniles) [inline server/controllers/finance/sale.js]: You could actually just delete this file - it is scheduled to be removed (see the comment in the file), and we never use it.

**Comment by jniles** [comment]
@DedrickEnc this is an extremely good refactor!

 If you want to delete `/finance/sale.js`, now would be a good time.  Otherwise, this is an extremely valuable refactor and I'll merge it as soon as possible.

Thanks!  You're a braver developer than I :)

**Comment by DedrickEnc** [comment]
The `finance/sale.js` file is now deleted.

@jniles @sfount

**Pull request merged**